lib/ci-helper.ts

```diff
@@ -763,6 +772,211 @@ export class CIHelper {
+                    } not found in ${pr.author}/${pr.baseRepo}.`);
+            }
+        } catch (e) {
+            if ((e as labelInfo).label === undefined ) {
+                const error = e as Error;
+                result = false;
+                await this.github.closePR(prInfo.url, `PR is being closed:\n\n${
+                    error.message}`);
+            } else {
+                const label = e as labelInfo;
+                await this.github.addPRLabels(prInfo.url, [label.label]);
+                await this.github.addPRComment(prInfo.url, `Label \`${
+                    label.label}\` added for this reason:\n\n${
+                    label.message}`);
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Perform some validation tests on a file in a pull request.
+     *
+     * @param changedFile file that is part of a commit in a pull request
+     * @param files number of files that are changed in the pull request
+     */
+    public checkNewUserFile(changedFile: PullRequestChangedFile,
+        files: number, setLabel: setLabelInfo):
+        void {
+        const filenameParts = path.parse(changedFile.path);
+
+        // one file with only deletions?
+        if (files === 1 &&
+            !changedFile.additions && changedFile.deletions) {
+            throw setLabel("validate", `File \`${changedFile.path
+                }\` expecting more than one change.`)
+        }
+
+        if (filenameParts.name.match(/.+\./)) {
+            throw new Error(`File \`${changedFile.path
+                }\` not allowed with multi-part name`);
+        }
+
+        if (filenameParts.name.match(/ /)) {
+            throw new Error(`File \`${changedFile.path
+                }\` not allowed with space in name`);
+        }
+
+        const badNames = ["SECURITY.md", "README"];
+        badNames.map(name => {
+            if (filenameParts.base.match(name)) {
+                throw setLabel("validate", `File \`${changedFile.path
+                    }\` not expected to be updated.`)
+            }
+        });
+
+        if (filenameParts.ext) {
+            if (filenameParts.ext.length > 11) { // allow .dockerfile?
+                throw new Error(`File \`${changedFile.path
+                    }\` has an invalid extension.`);
+            }
+
+            if (!filenameParts.ext.match(/^\.(png|ico|icns)$/)) {
+                if (!changedFile.additions && !changedFile.deletions) {
+                    throw new Error(`File \`${changedFile.path
+                        }\` not updated.`);
+                }
+            }
+
+            if (files === 1 &&
+                filenameParts.ext.match(/^\.([ch]|py|yml)$/)) {
+                // what if they rewrote some logic?
+                if (!changedFile.additions || !changedFile.deletions) {
+                    throw setLabel("validate", `File \`${changedFile.path
+                        }\` expecting more than one change.`)
```

Comment:
Maybe it would be sufficient to check for the addition of one empty line at the end of the file. IIRC that was the case in quite a couple bogus contributions.

Reply:
Depends upon whether we want to start reading files or running diffs.

Reply:
I'd really be in favor of reading the output of `git log -p <base-rev>..<tip-rev>`. That way, we can easily determine whether an empty line was added at the end of a file.